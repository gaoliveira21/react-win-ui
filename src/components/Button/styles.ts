import styled from 'styled-components'

export const Button = styled.button`
  border: none;
  background: red;
  color: #fff;
  cursor: pointer;
`
