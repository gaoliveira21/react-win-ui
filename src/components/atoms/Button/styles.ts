import styled, { css, DefaultTheme } from 'styled-components'

import { ButtonProps } from '.'

const modifiers = {
  primary: (theme: DefaultTheme) => css`
    color: ${theme.colors.text.ON_ACCENT};
    background-color: ${theme.colors.fill.accent.ACCENT_PRIMARY};

    &:hover {
      background-color: ${theme.colors.fill.accent.ACCENT_SECONDARY};
    }

    &:active {
      background-color: ${theme.colors.fill.accent.ACCENT_TERTIARY};
    }

    &:disabled {
      color: ${theme.colors.text.ON_ACCENT_DISABLED};
      background-color: ${theme.colors.fill.accent.DISABLED};
    }
  `,

  secondary: (theme: DefaultTheme) => css`
    color: ${theme.colors.text.PRIMARY};
    background-color: ${theme.colors.fill.control.CONTROL_PRIMARY};

    &:hover {
      background-color: ${theme.colors.fill.control.CONTROL_SECONDARY};
    }

    &:active {
      background-color: ${theme.colors.fill.control.CONTROL_TERTIARY};
    }

    &:disabled {
      color: ${theme.colors.text.DISABLED};
      background-color: ${theme.colors.fill.control.CONTROL_PRIMARY};
    }
  `
}

export const Button = styled.button<Pick<ButtonProps, 'variant'>>`
  ${({ theme, variant }) => css`
    font-size: ${theme.fonts.text.BODY.size};
    font-weight: ${theme.fonts.text.BODY.weight};
    line-height: ${theme.fonts.text.BODY.lineHeight};

    box-shadow: 0 0 2px 2px ${theme.colors.elevation.control.BORDER};

    padding: 5px 47px;
    border-radius: 4px;

    transition: background-color 0.3s;

    &:disabled {
      cursor: not-allowed;
    }

    &:focus {
      outline: 3px solid ${theme.colors.stroke.focus.OUTER};
      border: 1px solid ${theme.colors.stroke.focus.INNER};
    }

    ${!!variant && modifiers[variant](theme)};
  `}
`
