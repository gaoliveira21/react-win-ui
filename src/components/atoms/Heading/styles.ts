import styled, { css } from 'styled-components'

import { HeadingProps } from '.'

export const Heading = styled.div<Pick<HeadingProps, 'variant' | 'color'>>`
  ${({ theme, variant, color }) => css`
  font-family: ${theme.fonts.heading[variant.toUpperCase()].fontFamily};}
    font-size: ${theme.fonts.heading[variant.toUpperCase()].size};}
    font-weight: ${theme.fonts.heading[variant.toUpperCase()].weight};}
    line-height: ${theme.fonts.heading[variant.toUpperCase()].lineHeight};}
    color: ${theme.colors.text[color.toUpperCase()]};
  `}
`
