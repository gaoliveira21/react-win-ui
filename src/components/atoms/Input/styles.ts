import styled, { css } from 'styled-components'

export const Input = styled.input`
  ${({ theme }) => css`
    background-color: ${theme.colors.fill.control.CONTROL_PRIMARY};
    border: 1px solid ${theme.colors.elevation.textControl.BORDER};
    border-bottom: 2px solid ${theme.colors.stroke.control.SECONDARY};

    color: ${theme.colors.text.PRIMARY};
    font-size: ${theme.fonts.text.BODY.size};
    font-weight: ${theme.fonts.text.BODY.weight};
    line-height: ${theme.fonts.text.BODY.lineHeight};

    padding: 4px 35px 6px 11px;
    border-radius: 3px;

    &::placeholder {
      font-size: ${theme.fonts.text.BODY.size};
      font-weight: ${theme.fonts.text.BODY.weight};
      line-height: ${theme.fonts.text.BODY.lineHeight};
      color: ${theme.colors.text.SECONDARY};
    }

    &:focus {
      outline: none;
      border-bottom: 2px solid ${theme.colors.fill.accent.ACCENT_PRIMARY};
      background-color: ${theme.colors.fill.control.ACTIVE};
    }

    &:hover {
      &:not(:focus, :disabled) {
        background-color: ${theme.colors.fill.control.CONTROL_SECONDARY};
      }
    }

    &:disabled {
      background-color: ${theme.colors.fill.control.DISABLED};
      color: ${theme.colors.text.DISABLED};
      cursor: not-allowed;
    }
  `}
`
