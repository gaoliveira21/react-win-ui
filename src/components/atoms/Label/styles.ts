import styled, { css } from 'styled-components'

export const Label = styled.label`
  ${({ theme }) => css`
    font-size: ${theme.fonts.text.BODY.size};
    font-weight: ${theme.fonts.text.BODY.weight};
    line-height: ${theme.fonts.text.BODY.lineHeight};
    color: ${theme.colors.text.PRIMARY};
  `}
`
