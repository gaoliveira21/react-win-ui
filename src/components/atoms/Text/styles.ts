import styled from 'styled-components'

import { TextProps } from '.'

export const Text = styled.p<Pick<TextProps, 'variant'>>`
  font-family: ${({ theme, variant }) =>
    theme.fonts.text[variant.toUpperCase()].fontFamily};
  font-size: ${({ theme, variant }) =>
    theme.fonts.text[variant.toUpperCase()].size};
  font-weight: ${({ theme, variant }) =>
    theme.fonts.text[variant.toUpperCase()].weight};
  line-height: ${({ theme, variant }) =>
    theme.fonts.text[variant.toUpperCase()].lineHeight};
`
