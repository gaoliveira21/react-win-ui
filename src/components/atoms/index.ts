export * from './Text'
export * from './Heading'
export * from './Button'
export * from './Input'
export * from './Label'
