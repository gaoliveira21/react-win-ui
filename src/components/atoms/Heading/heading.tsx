import { TextTypes } from '@/styles/colors'
import { HeadingFonts } from '@/styles/typography'

import * as S from './styles'

export type HeadingProps = {
  children: React.ReactNode
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
  variant?: Lowercase<keyof HeadingFonts>
  color?: Lowercase<TextTypes>
}

export function Heading({
  children,
  as = 'h2',
  color = 'primary',
  variant = 'subtitle'
}: HeadingProps): React.ReactElement {
  return (
    <S.Heading as={as} color={color} variant={variant}>
      {children}
    </S.Heading>
  )
}
