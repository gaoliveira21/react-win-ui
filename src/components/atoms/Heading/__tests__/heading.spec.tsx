import { screen } from '@testing-library/react'

import { headingFonts } from '@/styles/typography'
import { darkTextColors, ligthTextColors } from '@/styles/colors'

import { renderWithTheme } from '@/tests/render-with-theme'

import { Heading, HeadingProps } from '@/components/atoms/Heading'

const assertFontStyle = (
  sut: Element,
  font: HeadingProps['variant'] = 'subtitle'
) => {
  expect(sut).toHaveStyle({
    'font-family': headingFonts[font.toUpperCase()].fontFamily,
    'font-size': headingFonts[font.toUpperCase()].size,
    'line-height': headingFonts[font.toUpperCase()].lineHeight,
    'font-weight': headingFonts[font.toUpperCase()].weight,
    color: ligthTextColors.PRIMARY
  })
}

describe('<Heading />', () => {
  it('Should render Heading with correct initial styles', () => {
    renderWithTheme(<Heading>text_content</Heading>)

    assertFontStyle(
      screen.getByRole('heading', { name: /text_content/i }),
      'subtitle'
    )
  })

  it('Should render Heading with correct text content', () => {
    const textContent = 'any_text'
    renderWithTheme(<Heading>{textContent}</Heading>)

    expect(
      screen.getByRole('heading', { name: textContent })
    ).toBeInTheDocument()
  })

  it('Should render Heading as a h2 by default', () => {
    renderWithTheme(<Heading>text_content</Heading>)

    expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument()
  })

  it('Should render Heading with different html element if "as" props has been passed', () => {
    renderWithTheme(<Heading as="h1">text_content</Heading>)

    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
  })

  it('Should render with "TITLE" styles if "variants" prop is equal to "title"', () => {
    renderWithTheme(<Heading variant="title">text_content</Heading>)

    assertFontStyle(
      screen.getByRole('heading', { name: /text_content/i }),
      'title'
    )
  })

  it('Should render with "TITLE_LARGE" styles if "variants" prop is equal to "title_large"', () => {
    renderWithTheme(<Heading variant="title_large">text_content</Heading>)

    assertFontStyle(
      screen.getByRole('heading', { name: /text_content/i }),
      'title_large'
    )
  })

  it('Should render with "DISPLAY" styles if "variants" prop is equal to "display"', () => {
    renderWithTheme(<Heading variant="display">text_content</Heading>)

    assertFontStyle(
      screen.getByRole('heading', { name: /text_content/i }),
      'display'
    )
  })

  it('Should render with correct color', () => {
    renderWithTheme(<Heading color="accent_primary">text_content</Heading>)

    expect(screen.getByRole('heading', { name: /text_content/i })).toHaveStyle({
      color: ligthTextColors.ACCENT_PRIMARY
    })
  })

  it('Should render with correct color if dark mode is enable', () => {
    renderWithTheme(<Heading color="accent_primary">text_content</Heading>, {
      dark: true
    })

    expect(screen.getByRole('heading', { name: /text_content/i })).toHaveStyle({
      color: darkTextColors.ACCENT_PRIMARY
    })
  })
})
