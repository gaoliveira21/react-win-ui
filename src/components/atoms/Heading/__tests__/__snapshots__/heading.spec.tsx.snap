// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> Should render Heading with correct initial styles 1`] = `
.c0 {
  font-family: SegoeUI;
  font-size: 20px;
  font-weight: 600;
  line-height: 28px;
  color: rgb(0,0,0);
}

<h2
  class="c0"
  color="primary"
>
  text_content
</h2>
`;
